test/unit-tests/core/vault-shares.ts

```diff
@@ -150,6 +150,13 @@ describe('Unit test: share calculating for pending deposit and withdraw', async
         const sharesToWithdraw = depositAmount
         await expect(vault.connect(depositor).redeem(sharesToWithdraw)).to.be.revertedWith('Exceeds available')
       })
+
+      it('should get 0 share if calling maxRedeem with no deposit Receipt', async() => {
+        const sharesBefore = await vault.balanceOf(anyone.address)
+        await vault.connect(anyone).maxRedeem()
+        const sharesAfter = await vault.balanceOf(anyone.address)
+        expect(sharesAfter).to.be.eq(sharesBefore)
+      })
   
```

Comment:
Let me know if I'm following the same logic: 
```typescript
 171 | it('should return 0 for shares before round ends', async() => {
        const balances = await vault.shareBalances(depositor.address)
        expect(balances.heldByVault).to.be.eq(0)
        expect(balances.heldByAccount).to.be.eq(0)
  
        const shares = await vault.shares(depositor.address)
        expect(shares).to.be.eq(0)
      })
```
yes, pricePerShare is still 0 (we have no placeholder so it's 0), but L172 won't revert because In `getSharesFromReceipt`,  it only call `assetToShares` if the depositReceipt.round < currentRound, so in this test, it will just return `unredeemedShares` which is zero
```typescript
  if (depositReceipt.round > 0 && depositReceipt.round < currentRound) {
    uint sharesFromRound = assetToShares(depositReceipt.amount, assetPerShare, decimals);

    return uint(depositReceipt.unredeemedShares).add(sharesFromRound);
  }
 return depositReceipt.unredeemedShares;
```

Reply:
I think they added this because people usually don't like view functions to revert, it could get quite annoying

Reply:
Sorry for the confusion but I was using the wrong code line -> I'm referring to L180
`      it('should return constant for price per share before first share is minted', async() => {
        const oneShare = parseEther('1')
        const price = await vault.pricePerShare()
        expect(price).to.be.eq(oneShare)
      })`

Reply:
it's because in `VaultLifecycle.rollover`
```typescript
50|  uint currentBalance = IERC20(asset).balanceOf(address(this));

     newPricePerShare = ShareMath.pricePerShare(currentShareSupply, currentBalance, pendingAmount, decimals);

     uint _mintShares = ShareMath.assetToShares(pendingAmount, newPricePerShare, decimals);
```
They're using the `currentShareSupploy` to calculate `newPricePerShare`, nad then using that to calculate how much shares (`_mintShares`) to mint.
 
if the current supply is 0, price should be default to 1, means you get exact shares based on your deposit. Otherwise _mintShares will be infinity (and causing the revert, cause price per share is 0).

That's why it should return 1 when there's no "shares" minted. not necessary when no assets has been deposited.

Reply:
Makes sense - in short, the first depositor should get "1" as price per share since no fee was taken and no profit was made yet